test/extended/images/hardprune.go

```diff
@@ -0,0 +1,242 @@
+			img, err := oc.AsAdmin().Client().Images().Get(imgName, metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			imgs[imgName] = img
+			o.Expect(img.DockerImageManifestMediaType).To(o.Equal(schema2.MediaTypeManifest))
+		}
+
+		deleted, err := RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		// this shouldn't delete anything
+		o.Expect(deleted).To(o.Equal([]string{}))
+
+		readOnly := true
+		err = registryutil.ConfigureRegistry(oc, registryutil.RegistryConfiguration{ReadOnly: &readOnly})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		/* imageName  | parent   | layers | imagestreams
+		 * ---------- | -------- | ------ | ------------
+		 *  baseImg1  |          | 1 2    | A Atagged
+		 *  baseImg2  |          | 4 5    | B
+		 *  baseImg3  |          | 7 8    | C
+		 *  baseImg4  |          | 11 12  | A
+		 *  childImg1 | baseImg1 | 1 2 3  | C
+		 *  childImg2 | baseImg2 | 4 5 6  | B
+		 *  childImg3 | baseImg3 | 7 8 9  | D
+		 */
+
+		// verify that the registry doesn't accept any images
+		_, _, err = BuildAndPushImageOfSizeWithDocker(oc, dClient, "fail", "tag", testImageSize, 1, outSink, false, true)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = oc.AsAdmin().Client().ImageStreamTags(oc.Namespace()).Delete("A", "latest")
+		deleted, err = RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = expectDeletions(deleted, oc.Namespace(),
+			expectedDeletions{
+				manifestLinks: repoLink{"A": []string{baseImg1}},
+				layerLinks: repoLink{"A": []string{
+					imgs[baseImg1].DockerImageMetadata.ID,
+					imgs[baseImg1].DockerImageLayers[0].Name,
+					imgs[baseImg1].DockerImageLayers[1].Name,
+				}},
+				blobs: []string{imgs[baseImg1].DockerImageMetadata.ID},
+			})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = oc.AsAdmin().Client().Images().Delete(childImg1)
+		deleted, err = RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = expectDeletions(deleted, oc.Namespace(),
+			expectedDeletions{
+				manifestLinks: repoLink{"C": []string{childImg1}},
+				layerLinks: repoLink{"C": []string{
+					imgs[childImg1].DockerImageMetadata.ID,
+					imgs[childImg1].DockerImageLayers[0].Name,
+				}},
+				blobs: []string{
+					childImg1,
+					imgs[childImg1].DockerImageMetadata.ID,
+					imgs[childImg1].DockerImageLayers[0].Name,
+				},
+			})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = oc.AsAdmin().Client().Images().Delete(baseImg1)
+		deleted, err = RunHardPrune(oc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = expectDeletions(deleted, oc.Namespace(),
+			expectedDeletions{
+				blobs: []string{
+					imgs[baseImg1].DockerImageLayers[0].Name,
+					imgs[baseImg1].DockerImageLayers[1].Name,
+				},
+			})
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// TODO continue to remove images
+	})
+
+})
+
+func expectDeletions(deletedPaths []string, namespace string, expect expectedDeletions) error {
```

Comment:
I'd like to read them from the standard output of `dockerregistry -prune` command. Will something like this be available? Or will it more like:

```
Deleted blob sha256:XXXX
Deleted layer link sha256:XXXX from repository NAMESPACE/REPO
```
?